front/src/modules/project/components/AddOrEditProjectModal.tsx

```diff
@@ -71,18 +81,36 @@ export default function AddOrEditProjectModal({
 
   const { updateProjectID } = useOsrdConfActions();
 
+  const initialValuesRef = useRef<ProjectForm | null>(null);
+
+  const handleInputChange = (field: string, value: string | number | string[] | undefined) => {
+    setCurrentProject((prevCurrentProject) => {
+      const updatedProject = {
+        ...prevCurrentProject,
+        [field]: value,
+      };
+
+      const hasValueChanged = !isEqual(updatedProject, initialValuesRef.current);
+      setHasChanges(hasValueChanged);
+
+      return updatedProject;
+    });
+  };
```

Comment:
this function is repeated in `AddOrEditStudyModal` and `AddOrEditScenarioModal`, it could be nice to isolate it in in a `utils` folder

Reply:
Ty ! Fixed